### PR TITLE
Trim whitespaces in object filter's library and object field

### PR DESCRIPTION
**Pull request by sebjulliand**
### Changes
Fixes https://github.com/codefori/vscode-ibmi/issues/1847

Whitespaces in object filter's `library` and `object` fields were not trimmed, resulting in the filtering not working correctly.

### Checklist
* [x] have tested my change

### CODE REVIEW

File: src/webviews/filters/index.ts [hunk 1]
```diff
@@ -66,22 +66,21 @@ export async function editFilter(filter?: ConnectionConfiguration.ObjectFilters,
...
             break;
           case `types`:
             data[key] = String(data[key]).split(`,`).map(item => item.trim().toUpperCase()).filter(item => item !== ``);
             break;
+          case `library`:
           case `object`:
             data[key] = (String(data[key].trim()) || `*`)
               .split(',')
-              .map(o => useRegexFilters ? o : o.toLocaleUpperCase())
+              .map(o => useRegexFilters ? o : o.toLocaleUpperCase().trim())
```
Comment (chrjorgensen) [inline src/webviews/filters/index.ts:79]: @sebjulliand Why trimming filters when saving the filter? The filter works great with or without blanks in-between, and not trimming the filters allows the user to space the filter out for better readability...

**Review by chrjorgensen — CHANGES_REQUESTED** [review]
@sebjulliand Code looks good to me - only one question... 😉

File: src/webviews/filters/index.ts (hunk 1 above)
Comment (chrjorgensen) [reply src/webviews/filters/index.ts:79]: Maybe just trimming before splitting would be sufficient - then the user can have spaces in-between, and the filter would still be left-justified without trailing spaces.

**Comment by worksofliam** [comment]
Can we get a test case for this? :)

File: src/webviews/filters/index.ts (hunk 1 above)
Comment (sebjulliand) [reply src/webviews/filters/index.ts:79]: That may indeed be better! Thanks Christian!

**Comment by sebjulliand** [comment]
@chrjorgensen filters are not trimmed on saving anymore
@worksofliam I added a test for filters with whitespaces

**Review by chrjorgensen — APPROVED** [review]
@sebjulliand You rock :rocket: :+1: - and work in the weekend! :-1: :wink: 

But so do I, so it's okay... :laughing: 

Approved and merged. We should think of making a prerelease for users to avoid this error...

**Pull request merged**